subsys/bluetooth/audio/has.c

```diff
@@ -34,7 +56,362 @@ static ssize_t read_features(struct bt_conn *conn, const struct bt_gatt_attr *at
+{
+	preset->index = BT_HAS_PRESET_INDEX_NONE;
+}
+
+static void control_point_ntf_complete(struct bt_conn *conn, void *user_data)
+{
+	struct has_client *client = client_get(conn);
+
+	BT_DBG("conn %p\n", (void *)conn);
+
+	/* Resubmit if needed */
+	if (read_presets_req_is_pending(client)) {
+		k_work_submit(&client->control_point_work);
+	}
+}
+
+static void control_point_ind_complete(struct bt_conn *conn,
+				       struct bt_gatt_indicate_params *params,
+				       uint8_t err)
+{
+	if (err) {
+		/* TODO: Handle error somehow */
+		BT_ERR("conn %p err 0x%02x\n", (void *)conn, err);
+	}
+
+	control_point_ntf_complete(conn, NULL);
+}
+
+static int control_point_send(struct has_client *client, struct net_buf_simple *buf)
+{
+#if defined(CONFIG_BT_EATT)
+	/* TODO: Check whether EATT is connected */
+	if (bt_gatt_is_subscribed(client->conn, &has_svc.attrs[4], BT_GATT_CCC_NOTIFY)) {
+		client->params.ntf.attr = &has_svc.attrs[4];
+		client->params.ntf.func = control_point_ntf_complete;
+		client->params.ntf.data = buf->data;
+		client->params.ntf.len = buf->len;
+
+		return bt_gatt_notify_cb(client->conn, &client->params.ntf);
+	}
+#endif /* CONFIG_BT_EATT */
+
+	if (bt_gatt_is_subscribed(client->conn, &has_svc.attrs[4], BT_GATT_CCC_INDICATE)) {
+		client->params.ind.attr = &has_svc.attrs[4];
+		client->params.ind.func = control_point_ind_complete;
+		client->params.ind.destroy = NULL;
+		client->params.ind.data = buf->data;
+		client->params.ind.len = buf->len;
+
+		return bt_gatt_indicate(client->conn, &client->params.ind);
+	}
+
+	return -ECANCELED;
+}
+
+static int bt_has_cp_read_preset_rsp(struct has_client *client, struct bt_has_preset *preset,
+				     uint8_t is_last)
+{
+	struct bt_has_cp_hdr *hdr;
+	struct bt_has_cp_read_preset_rsp *rsp;
+
+	NET_BUF_SIMPLE_DEFINE(buf, sizeof(*hdr) + sizeof(*rsp) + BT_HAS_PRESET_NAME_MAX);
+
+	BT_DBG("conn %p preset %p is_last 0x%02x", (void *)client->conn, preset, is_last);
+
+	hdr = net_buf_simple_add(&buf, sizeof(*hdr));
+	hdr->opcode = BT_HAS_OP_READ_PRESET_RSP;
+	rsp = net_buf_simple_add(&buf, sizeof(*rsp));
+	rsp->is_last = is_last;
+	rsp->index = preset->index;
+	rsp->properties = preset->properties;
+	net_buf_simple_add_mem(&buf, preset->name, strlen(preset->name));
+
+	return control_point_send(client, &buf);
+}
+
+static uint8_t preset_find_next(const struct bt_has_preset *preset, void *user_data)
+{
+	const struct bt_has_preset **found = user_data;
+
+	if (*found == NULL || (*found)->index > preset->index) {
+		*found = preset;
+	}
+
+	return BT_HAS_PRESET_ITER_CONTINUE;
+}
+
+static void process_control_point_work(struct k_work *work)
+{
+	struct has_client *client = CONTAINER_OF(work, struct has_client, control_point_work);
+	int err;
+
+	if (!client->conn) {
+		return;
+	}
+
+	if (read_presets_req_is_pending(client)) {
+		struct bt_has_preset *preset = NULL;
+		const uint8_t start_index = client->read_presets_req.start_index;
+		uint8_t is_last;
```

Comment:
I know this value is being converted to a `uint8_t` in the response, but in this context (and the argument of `bt_has_cp_read_preset_rsp`), I think it would make more sense to use the `bool` type, and then in `bt_has_cp_read_preset_rsp` do 
```c
rsp->is_last = is_last ? 0x01 : 0x00;
```